docs/make.jl

```diff
@@ -62,6 +62,7 @@ bibtex_plugin = CitationBibliography(
         "Topic guides" => [
             "Topic guide overview" => "topics/index.md",
             "topics/fe_intro.md",
+            "topics/reference_shapes.md",
```

Comment:
There is `devdocs/reference_cell.md` where we refer to these as "reference cells" rather than "reference shapes". I think they are always synonyms in our case though, right?

Reply:
Yes. But they are `AbstractRefShape`, so shape makes more sense to me here? What do you think?

Of course, per definition now the `Cell` is the `RefShape<:AbstractRefShape` since it is codim 0 ...

Reply:
Note that there is still a slight difference (which we might want to point out at least in the devdocs). The reference shape is for us really the integration domain in standardized coordinates, which the cell is technically the reference domain mapped into the physical domain by connecting it with a geometric Ansatz function and associated coordinates.

Reply:
Good point, 
do you think the current [topics doc](https://ferrite-fem.github.io/Ferrite.jl/previews/PR935/topics/reference_shapes/) is off in this respect?

Reply:
A thing which I spotted is that vertex refers to point, not to corner.

Reply:
Thanks!

> A thing which I spotted is that vertex refers to point, not to corner.

Yes, chose avoiding corner since calling the endpoints of a `Line` for corner sounds strange to me.

> No, but it does not explain it specifically either.

Good, that should be described in either FEValues or Grid IMO.

Reply:
> Thanks!
> 
> > A thing which I spotted is that vertex refers to point, not to corner.
> 
> Yes, chose avoiding corner since calling the endpoints of a `Line` for corner sounds strange to me.
> 

Good point. But this also creates an ambiguity with the nodes used to describe the nonlinearity of some cells. Any idea if we can resolve this?

> > No, but it does not explain it specifically either.
> 
> Good, that should be described in either FEValues or Grid IMO.

Actually this should be in the docs for the cells/elements maybe? And then the other docs can refer to this. This is especially important for new users to understand that we can decouple the geometry interpolation from the solution interpolation, and that this trick comes with some issues e.g. for visualization in ParaView. Probably this train of thought should be put into some docs PR along other missing things.

Reply:
OK, I added a short note about the difference between a node and a vertex (this could be improved with a link if we have another place to describe the node concept). 

Especially with #798, there will be a need to explain non-nodal interpolations.

But I'll suggest leaving that for later. I think the docs in this PR is required before 1.0, but the others can be done later.

Reply:
> reference cells" rather than "reference shapes"

For interpolations we also refer to them as "reference elements" 😆

Reply:
Was probably me haha

Reply:
I don't want to add a lot of rename to this PR, but focus on getting the new notation documented. IMO we should go over the docs and use "reference shape" consistently in another PR. I'll open (yet another issue) for that, so that we don't forget (also a nice first-time pr if we reach consensus in the issue discussion)

Reply:
Sure, let us split these issues into smaller ones and just keep track of them like you propose. Thanks.